### DISCUSSION TITLE
Environment Variables and Parameters Available

### DISCUSSION THREAD

**Discussion by ghost** [intersystems/ipm#395] [Q&A]
Hi,

I am looking into the FileCopy part of the module.xml file
I see on the wiki the following in the example:
<FileCopy Name="somefile.jar" Target="${libdir}_my-lib/"/>
My questions is regarding $libdir. Where is libdir defined and assigned a value?
Is it in the OS environment? I want to use a variable for the location of some files, which may differ per environment where the package will be loaded.
I would like some information on the use of these variables, which are available if not read from the OS env, can I add more, etc.
The documentation covers basics and finding information on functionality is not easy.

Thanks

**Comment by daimor** [comment]
Hi, 
You can look at the source code, where it evaluated

https://github.com/intersystems/ipm/blob/5a5f5f428667d353393e5254a5fc5a63f175d14d/src/%25ZPM/PackageManager/Developer/Module.cls#L1426

**Comment by evshvarov** [comment]
Indeed we need more examples. Could yuo pleese submit an issue?
E.g ${libdir} is crucial to deploy embedded python solutions
Here is how it is being used
https://github.com/evshvarov/csvgen-python/blob/7680248aea622947575e0267519c00512fb8264f/module.xml#L11
<FileCopy Name="app/" Target="${libdir}python/"/>
to make it then called as:
Set csv = ##class(%SYS.Python).Import("csvgen")
Source
https://github.com/evshvarov/csvgen-python/blob/7680248aea622947575e0267519c00512fb8264f/src/shvarov/csvgenpy/generate.cls#LL9C9-L9C56

**Comment by evshvarov** [comment]
Also, Stefan, you are very welcome to PR for the features you are
especially interested in!
This is an open source lib!
